### ISSUE TITLE
Weakly performance on real dataset?

### ISSUE THREAD

**Issue by AbyssGaze** [NeurAI-Lab/biHomE#7]
I have trained pseudo homography dataset(augmented with a photometrical distortion), by when I evaluated with a real dataset, the performance is weakly(the smaller the better).
![image](https://user-images.githubusercontent.com/14217667/156715290-425bfed9-5639-4500-a9b5-b51b92c04c40.png)

**Comment by dkoguciuk** [comment]
Hi @AbyssGaze ,

sorry for the late reply. Can you help me understand the metrics? I can see that training MACE is around 1.15, but what is the metric on the real dataset?

In general, I've noticed, that homography estimation models trained with perceptual loss are sensitive to the data domain shift. For example, I've trained the model on COCO and then tested it on Oxford RobotCar Dataset and the performance was moderate compared to the model trained and tested on Oxford. If possible, try to learn the biHomE on your target dataset (you don't need any labels for that of course).

Best,
D

P.S. Congratulations on your AAAI2022 paper :)

**Comment by AbyssGaze** [comment]
I'm very sorry for replying to your message so late. I tested and trained on the same source data. The only difference is that the training data is constructed after augmentation, and the inference data is the real labeled data. Due to sensitive data, I may not be able to show you the corresponding visualization results for the time being sorry.

**Comment by dkoguciuk** [comment]
Gu @AbyssGaze ,

it's really hard for me to say what could be the problem here, but we can play around a bit. Can you try to learn the model once more, but without augmentation on the trainset? Also, you could try to check other unsupervised methods like Nguyen and Zhang (both implemented in this repo), then we could narrow down the problem to either the biHomE method or the data/convergence problem.

Best,
D

**Comment by AbyssGaze** [comment]
Thank you for your prompt reply.
I have uploaded the log file of the relevant training results, the visualization of some data, and the original pair data to the cloud disk: https://drive.google.com/drive/folders/1jnl2v3iFoXgXwRkzcen4PgxPbbjENkVA?usp=sharing. 

Judging from the training log file **bihome_log.zip** and the corresponding visualization result **bihome_aug_viz**, it can converge well on the pseudo data. Still, the pair under **origin_example** does not run well on the data, and this data is also simple for image registration. Some data inference visualizations are shown in **real_data_visualization**.

Thank you for your assistance.

**Comment by dkoguciuk** [comment]
Hi @AbyssGaze ,

thanks for the data upload. Let me ask some more questions to make sure that I understand it correctly. 

1. `bihome_aug_viz` is the visualization of the model trained using augmentations - so basically the same procedure that was used to produce S-COCO or PDS-COCO. Is that right?
2. `real_data_visualization` is the visualization of the model trained using **real pair of images** -- no synthetic data creation here was used, right?
3. What is the difference between the `zeng-bihome-pdscoco-lr-1e-3` model and the `bihome_log` model? I can see that one of them is trained for 180k epochs and one for 90k epochs, but what is exactly the bihome_log model? Is it also zeng+bihome? And on what data they were trained? 

Best,
Daniel

**Comment by AbyssGaze** [comment]
Hi Daniel, 
1. `bihome_aug_viz`  is the inference results visualization of augmentation datasets, the same procedure used to produce S-COCO or PDS-COCO for our datasets as presented in `origin_example`, with the checkpoint of `bihome_log/model_180000.pth`. 
2. `real_data_visualization`  is with visualization of  the *real pair of images* inferenced results of  `bihome_log/model_180000.pth`.
3. `zeng-bihome-pdscoco-lr-1e-3` used default config for PDS-COCO training,  and `bihome_log` is trained with the origin datasets presented in `origin_example` with the same augmentation of PDS-COCO program, and the origin model name is zeng-bihome-pdhomo-lr-1e-3-origin-new-400.
Best,
mumu

**Comment by dkoguciuk** [comment]
Hi @AbyssGaze ,

Ok, now I understand, thanks for the clarification. I think that the problem is that homography is not the best model to register those images. Homography can be used for image registration only in two cases: the scene is planar or the camera undergoes rotational movement. Here, both assumptions are violated as far as I understand.

I always wanted to try one thing: the Zeng model estimates per pixel delta (like optical flow) and in the paper, I have applied the RANSAC model fitting to make I work for homography. But what if you use this perspective field estimation without homography in the inference mode, maybe you would be able to non-rigidly transform one of the images onto another? 

Best,
D

**Comment by dkoguciuk** [comment]
I will close the issue for now but feel free to re-open it if you want to discuss anything :)
